ogre2/src/Ogre2RayQuery.cc

```diff
@@ -56,6 +56,12 @@ class gz::rendering::Ogre2RayQueryPrivate
 
   /// \brief thread that ray query is created in
   public: std::thread::id threadId;
+
+  public: ~Ogre2RayQueryPrivate() {
+    if (rayQuery != nullptr) {
+      delete rayQuery;
+    }
```

Comment:
how about putting this in the `Ogre2RayQuery` destructor? and use ogre's scene manager to destroy the query object:

```diff
diff --git a/ogre2/src/Ogre2RayQuery.cc b/ogre2/src/Ogre2RayQuery.cc
index 8d7684c9..8a9826c3 100644
--- a/ogre2/src/Ogre2RayQuery.cc
+++ b/ogre2/src/Ogre2RayQuery.cc
@@ -71,6 +71,16 @@ Ogre2RayQuery::Ogre2RayQuery()
 //////////////////////////////////////////////////
 Ogre2RayQuery::~Ogre2RayQuery()
 {
+  if (this->dataPtr->rayQuery)
+  {
+    Ogre2ScenePtr ogreScene =
+        std::dynamic_pointer_cast<Ogre2Scene>(this->Scene());
+    if (ogreScene)
+    {
+      ogreScene->OgreSceneManager()->destroyQuery(this->dataPtr->rayQuery);
+      this->dataPtr->rayQuery = nullptr;
+    }
+  }
 }
```